webapp/src/main/resources/redis/scripts/sliding-window-rate-limiter.lua

```diff
@@ -0,0 +1,26 @@
+-- Sliding Window Rate Limiter
+local key = KEYS[1]                         -- Rate limit key (e.g. "ai_translate", "search")
+local max_requests = tonumber(ARGV[1])      -- Maximum requests allowed in the time window e.g 300
+local window_size_ms = tonumber(ARGV[2])    -- Time window in milliseconds e.g 60000 (1 minute)
+
+-- Get the current time in milliseconds
+local now = redis.call('TIME') -- Returns current time as an array [seconds, microseconds elapsed in the current second]
+local now_ms = now[1] * 1000 + math.floor(now[2] / 1000)
+
+-- Requests are stored with their current time as scores
+-- Remove all entries with scores between 0 and (current_time - window_size)
+redis.call('ZREMRANGEBYSCORE', key, 0, now_ms - window_size_ms)
```

Comment:
Interesting. I thought the cache items would clear themselves if you set a TTL on the key and that you wouldn't need to do this manually